sacc/tracers.py

```diff
@@ -19,16 +36,23 @@ class BaseTracer:
     """
     _tracer_classes = {}
 
-    def __init__(self, name, **kwargs):
+    def __init__(self, name, quantity='generic', **kwargs):
```

Comment:
This breaks existing code like this in TXPipe:
`S.add_tracer('NZ', f'source_{i}', z, Nz)`
since the name is interpreted as being `quantity`.
You can instead do e.g. `quantity=kwargs.pop('quantity', 'generic')`

Reply:
OK @joezuntz, happy to do that, but this will stay broken because now NZ tracers also require a `spin` parameter. Thoughts?

Reply:
It shouldn't require a `spin`.  In many cases we will use the same n(z) for both spin 2 and spin 0 quantities, e.g. 3x2pt with the same lens and source bins.  An optional one makes sense, as with `quantity`.

Reply:
Aside from the spin issue (which I'm happy to make optional), I think this would be very bad practice. I don't think it'll ever be the case that sources and lenses will have exactly the same N(z)s, since galaxy weights will be different for both.

Reply:
In any case, spin is now an optional tag.

Reply:
so no previous code is broken by this.